swap/src/monero.rs

```diff
@@ -42,6 +42,13 @@ pub fn private_key_from_secp256k1_scalar(scalar: bitcoin::Scalar) -> PrivateKey
 #[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct PrivateViewKey(#[serde(with = "monero_private_key")] PrivateKey);
 
+impl fmt::Display for PrivateViewKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Delegate to the Display implementation of PrivateKey
+        write!(f, "{}", self.0)
+    }
+}
```

Comment:
why is this necessary? it's a bit of a footgun to implement Display like this for secrets. What I've seen in a crate like rust-secp256k1 is that the Display impl still hides the secret, and it's necessary to call a very specific `unsafe_display_secret` or something if you want to view the actual secret.  

https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/secret.rs#L66

Reply:
I did this so that we can format the private key properly to be printed to the console / API caller.

I'll modify it to use the display_secret method instead. I didn't know that existed, thanks :)

Reply:
I took a look again and I think I'm really just continuing how the display of secret was treated before.

```rust
let (spend_key, view_key) = state5.xmr_keys();

                    let address = monero::Address::standard(
                        context.config.env_config.monero_network,
                        monero::PublicKey::from_private_key(&spend_key),
                        monero::PublicKey::from(view_key.public()),
                    );

                    tracing::info!(address=%address, spend_key=%spend_key, view_key=view_key.to_secret_string(), "Monero recovery information");

                    Ok(json!({
                        "address": address,
                        "spend_key": spend_key.to_string(),
                        "view_key": view_key.to_secret_string(),
                    }))
```

There are two types of Monero keys. Spend key (which is of type PrivateKey) and ViewKey which is of type (PrivateViewKey). PrivateKey implements Display already. PrivateViewKey is basically just a wrapped ViewKey with some extra traits. The code I added just calls that underlying Display implementation.

Therefore, the current convention in this project seems to be implement display even for secrets. SpendKey is arguably more secret than ViewKey and already implements Display so why shouldn't ViewKey as well?

Do you agree or should I remove the Display implementation for all private key structs. Seems out of scope for this PR to me imo.